lib/resources.js

```diff
@@ -276,23 +276,48 @@ module.exports = function (app, _private = null) {
       .then((items) => ItemResultsSerializer.serialize(items, opts))
   }
 
+  const addInnerHits = (body) => {
+    body.query.function_score.query.bool.should.push({
+      nested: {
+        path: 'items',
+        query: { 'match_all': {} },
+        inner_hits: {
+          sort: [{ 'shelfMark_sort': 'asc' }],
+          size: process.env.SEARCH_ITEMS_SIZE
+        }
+      }
+    })
+    return body
+  }
+
   // Conduct a search across resources:
   app.resources.search = function (params, opts, request) {
     params = parseSearchParams(params)
 
     var body = buildElasticBody(params)
 
+    const shouldAddInnerHits = body.query && body.query.function_score && body.query.function_score.query && body.query.function_score.query.bool && body.query.function_score.query.bool.should
```

Comment:
Did you find that the query failed when you added inner_hits but there was no active boolean query?

Reply:
Some of the tests fail without this. I think the `search` function is used by requests that have different body shapes. I can look further into why the tests fail without this.

Reply:
Looks like some of the tests are doing empty searches (where the resulting elasticsearch query does not include the `body.query.function_score.query.bool.should` structure). If your `addInnerHits` function depends on that structure being there, it will fail. The risk of simply not adding the inner_hits query for empty searches is that one of the items returned from that empty search may have excessive items, triggering the bug. It's not likely, but it's possible. I wonder what you think of removing the `shouldAddInnerHits` check you have in favor of dynamically adding in the necessary structure as needed _within_ `addInnerHits`, like:

```
  const addInnerHits = (body) => {
    // Ensure existing ES body has structure for bool query:
    if (!body.query) body.query = {}
    if (!body.query.function_score) body.query.function_score = {}
    if (!body.query.function_score.query) body.query.function_score.query = {}
    if (!body.query.function_score.query.bool) body.query.function_score.query.bool = {}
    if (!body.query.function_score.query.bool.should) body.query.function_score.query.bool.should = []
    ...
```

That's not pretty (and you could make it much shorter by relying on some assumptions about what kind of ES query is being built upstream - or using some kind of recursive merge util), but it ensures your inner_hits fix applies to all queries however they're constructed.

Reply:
Also, I gather the fixtures are changing frequently, but looks like this PR doesn't include the new fixtures.

Reply:
1. Regarding adding the `body.query.function_score.query.bool.should` structure: I would be fine with that, except I feel like I don't understand elasticsearch well enough to know whether adding that could have any undesirable effects on other queries. If you are confident that it won't, I'd be happy to go that route. It doesn't seem we get any timeouts with empty searches, but I can't say for sure that that will be true for all param values/in the future
2. It looks to me that github is just rendering a message for each of the new fixtures that large diffs are not rendered by default. Is that not what you are seeing?

Reply:
1. Since this is a bug fix, fair enough to play it safe. We don't believe the bug effects other queries, so let's not change them just yet.
2. Oddly when running `npm test` locally, I get errors about missing fixtures on 15 tests. You don't have any local fixtures that aren't committed?

Reply:
That is strange. I am getting all passing tests and clean git status